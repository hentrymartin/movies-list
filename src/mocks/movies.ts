export const Movie = {
  "popularity": 136.328,
  "id": 529149,
  "video": false,
  "vote_count": 11,
  "vote_average": 4,
  "title": "Darlin'",
  "release_date": "2020-07-10",
  "original_language": "en",
  "original_title": "Darlin'",
  "genre_ids": [
    27,
    9648
  ],
  "backdrop_path": "/k60vmF4t3X9Yw4oAiQ2s0KtN7RU.jpg",
  "adult": false,
  "overview": "Found at a Catholic hospital filthy and ferocious, feral teenager Darlin’ is whisked off to a care home run by The Bishop and his obedient nuns, where she’s to be rehabilitated into a \"good girl\" as an example of the miraculous work of the church. But Darlin’ holds a secret darker than the \"sins\" she is threatened with, and she is not traveling alone. The Woman who raised her, equally fierce and feral, is ever present in the shadows of Darlin’s psyche and is determined to come for her no matter who tries to get in her way.",
  "poster_path": "/hnNjlXAQan8WdNtQMfdldAGoqMF.jpg"
};

export const Movies = [Movie,
{
  "popularity": 114.677,
  "vote_count": 153,
  "video": false,
  "poster_path": "/4WT7zYFpe0fsbg6TitppiHddWAh.jpg",
  "id": 75624,
  "adult": false,
  "backdrop_path": null,
  "original_language": "ja",
  "original_title": "劇場版 NARUTO -ナルト- ブラッド・プリズン",
  "genre_ids": [
    28,
    16,
    35,
    27,
    9648,
    53
  ],
  "title": "Naruto Shippuden the Movie: Blood Prison",
  "vote_average": 6.9,
  "overview": "After his capture for attempted assassination of the Raikage, leader of Kumogakure, as well as killing Jōnin from Kirigakure and Iwagakure, Naruto is imprisoned in Hōzukijou: A criminal containment facility known as the Blood Prison. Mui, the castle master, uses the ultimate imprisonment technique to steal power from the prisoners, which is when Naruto notices his life has been targeted. Thus begins the battle to uncover the truth behind the mysterious murders and prove Naruto's innocence.",
  "release_date": "2011-07-30"
},
{
  "popularity": 105.114,
  "vote_count": 477,
  "video": false,
  "poster_path": "/yf5IuMW6GHghu39kxA0oFx7Bxmj.jpg",
  "id": 587792,
  "adult": false,
  "backdrop_path": "/d7JUXVvjvVCXWs1mlpyO5ESdWdT.jpg",
  "original_language": "en",
  "original_title": "Palm Springs",
  "genre_ids": [
    35,
    14,
    9648,
    10749
  ],
  "title": "Palm Springs",
  "vote_average": 7.6,
  "overview": "When carefree Nyles and reluctant maid of honor Sarah have a chance encounter at a Palm Springs wedding, things get complicated when they find themselves unable to escape the venue, themselves, or each other.",
  "release_date": "2020-07-10"
},
{
  "popularity": 95.111,
  "vote_count": 546,
  "video": false,
  "poster_path": "/tIpGQ9uuII7QVFF0GHCFTJFfXve.jpg",
  "id": 555974,
  "adult": false,
  "backdrop_path": "/rpGYHowXtjw37UxdwO1ZcK5E8IN.jpg",
  "original_language": "en",
  "original_title": "Brahms: The Boy II",
  "genre_ids": [
    27,
    9648,
    53
  ],
  "title": "Brahms: The Boy II",
  "vote_average": 6.3,
  "overview": "After a family moves into the Heelshire Mansion, their young son soon makes friends with a life-like doll called Brahms.",
  "release_date": "2020-02-20"
},
{
  "popularity": 84.394,
  "vote_count": 1899,
  "video": false,
  "poster_path": "/qWsHMrbg9DsBY3bCMk9jyYCRVRs.jpg",
  "id": 521029,
  "adult": false,
  "backdrop_path": "/jB98SrdXAYSbiprjIwc7WfVCuCV.jpg",
  "original_language": "en",
  "original_title": "Annabelle Comes Home",
  "genre_ids": [
    27,
    9648,
    53
  ],
  "title": "Annabelle Comes Home",
  "vote_average": 6.4,
  "overview": "Determined to keep Annabelle from wreaking more havoc, demonologists Ed and Lorraine Warren bring the possessed doll to the locked artifacts room in their home, placing her “safely” behind sacred glass and enlisting a priest’s holy blessing. But an unholy night of horror awaits as Annabelle awakens the evil spirits in the room, who all set their sights on a new target—the Warrens' ten-year-old daughter, Judy, and her friends.",
  "release_date": "2019-06-26"
},
{
  "popularity": 78.543,
  "vote_count": 114,
  "video": false,
  "poster_path": "/chHECAPZTPydk8NHVCo9irCcH24.jpg",
  "id": 65650,
  "adult": false,
  "backdrop_path": "/pRMZilt1YIsu6aNnlpf0lo5Brm4.jpg",
  "original_language": "en",
  "original_title": "The Good Doctor",
  "genre_ids": [
    18,
    9648,
    53
  ],
  "title": "The Good Doctor",
  "vote_average": 5.8,
  "overview": "Dr. Martin Blake, who has spent his life looking for respect, meets an 18-year-old patient named Diane, suffering from a kidney infection, and gets a much-needed boost of self-esteem. However, when her health starts improving, Martin fears losing her, so he begins tampering with her treatment, keeping Diane sick and in the hospital right next to him.",
  "release_date": "2011-04-23"
},
{
  "popularity": 74.161,
  "vote_count": 4098,
  "video": false,
  "poster_path": "/sFC1ElvoKGdHJIWRpNB3xWJ9lJA.jpg",
  "id": 439079,
  "adult": false,
  "backdrop_path": "/fgsHxz21B27hOOqQBiw9L6yWcM7.jpg",
  "original_language": "en",
  "original_title": "The Nun",
  "genre_ids": [
    27,
    9648,
    53
  ],
  "title": "The Nun",
  "vote_average": 5.8,
  "overview": "When a young nun at a cloistered abbey in Romania takes her own life, a priest with a haunted past and a novitiate on the threshold of her final vows are sent by the Vatican to investigate. Together they uncover the order’s unholy secret. Risking not only their lives but their faith and their very souls, they confront a malevolent force in the form of the same demonic nun that first terrorized audiences in “The Conjuring 2” as the abbey becomes a horrific battleground between the living and the damned.",
  "release_date": "2018-09-05"
},
{
  "popularity": 71.61,
  "vote_count": 13462,
  "video": false,
  "poster_path": "/g6n8G0xPnBVUIKKK41sNmbZB5h4.jpg",
  "id": 675,
  "adult": false,
  "backdrop_path": "/pkxPkHOPJjOvzfQOclANEBT8OfK.jpg",
  "original_language": "en",
  "original_title": "Harry Potter and the Order of the Phoenix",
  "genre_ids": [
    12,
    14,
    9648
  ],
  "title": "Harry Potter and the Order of the Phoenix",
  "vote_average": 7.7,
  "overview": "Returning for his fifth year of study at Hogwarts, Harry is stunned to find that his warnings about the return of Lord Voldemort have been ignored. Left with no choice, Harry takes matters into his own hands, training a small group of students – dubbed 'Dumbledore's Army' – to defend themselves against the dark arts.",
  "release_date": "2007-06-28"
},
{
  "popularity": 66.58,
  "vote_count": 12470,
  "video": false,
  "poster_path": "/ode14q7WtDugFDp78fo9lCsmay9.jpg",
  "id": 198663,
  "adult": false,
  "backdrop_path": "/vHJlbhsXrZ5yrO2KqCbGSIU6fRX.jpg",
  "original_language": "en",
  "original_title": "The Maze Runner",
  "genre_ids": [
    28,
    9648,
    878,
    53
  ],
  "title": "The Maze Runner",
  "vote_average": 7.1,
  "overview": "Set in a post-apocalyptic world, young Thomas is deposited in a community of boys after his memory is erased, soon learning they're all trapped in a maze that will require him to join forces with fellow “runners” for a shot at escape.",
  "release_date": "2014-09-10"
},
{
  "popularity": 65.328,
  "vote_count": 649,
  "video": false,
  "poster_path": "/vN7JHlHOT9rHNDU27tfYqhABBj5.jpg",
  "id": 465086,
  "adult": false,
  "backdrop_path": "/46a1JNGmCSfoimlxtuxeDquuQ37.jpg",
  "original_language": "en",
  "original_title": "The Grudge",
  "genre_ids": [
    27,
    9648
  ],
  "title": "The Grudge",
  "vote_average": 5.7,
  "overview": "After a young mother murders her family in her own house, a detective attempts to investigate the mysterious case, only to discover that the house is cursed by a vengeful ghost. Now targeted by the demonic spirits, the detective must do anything to protect herself and her family from harm.",
  "release_date": "2020-01-02"
}];
