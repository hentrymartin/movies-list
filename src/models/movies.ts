import {
  types,
  cast,
  flow,
  applySnapshot,
  } from "mobx-state-tree";
import { getMoviesListByGenreId } from "../api/api";


  export const Movie = types.model({
    poster_path: types.maybeNull(types.string),
    popularity: types.number,
    original_title: types.string,
    title: types.string,
    vote_average: types.number,
    overview: types.string,
    release_date: types.string,
    genre_ids: types.array(types.number),
    original_language: types.string,
    backdrop_path: types.maybeNull(types.string),
    adult: types.boolean,
    id: types.number,
    media_type: types.maybe(types.string),
    video: types.boolean,
    vote_count: types.number,
  });

  export const GenreMetadata = types.model({
    id: types.number,
    name: types.string,
  });

  export const MoviesList = types.model({
    list: types.optional(types.array(Movie), []),
    name: types.string,
    page: types.number,
    loading: types.boolean,
  })
  .actions(self => ({
    getMoviesById: flow(function* (page: number, id?: number) {
      self.loading = true;
      
      // for the 1st page we are just resetting the list to empty array
      if (page === 1) {
        applySnapshot(self.list, []);
      }

      try {
        const response = yield getMoviesListByGenreId(page, id);
        const { results } = yield response.json();
        applySnapshot(self.list, self.list.concat(results));
      } catch(e) {
        console.log(e);
      }
      self.loading = false;
      self.page = page;
    }),
    resetValues() {
      self.page = 1;
      applySnapshot(self.list, []);
    },
    setGenreName(name: string) {
      self.name = name;
    },
  }));