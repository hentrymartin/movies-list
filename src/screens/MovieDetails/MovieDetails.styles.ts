import styled from 'styled-components/native';

export const MovieDetailsWrapper = styled.View`
  padding: 10px 0;
`;

export const MovieDetailsTitle = styled.Text`
  font-size: 20px;
  font-weight: 500;
  text-align: center;
  margin: 10px 0;
`;

export const MovieDescription = styled.Text`
  padding: 10px 20px;
  line-height: 22px;
  color: grey;
`;

export const MenuWrapper = styled.View`
  display: flex;
  padding: 0 20px;
  align-items: flex-end;
  justify-content: center;
`;
