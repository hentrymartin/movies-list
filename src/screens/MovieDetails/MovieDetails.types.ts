import { StackScreenProps } from '@react-navigation/stack';
import { MovieItem } from '../MoviesListView/MoviesListView.types';

export type MovieDetailsProps = {
  route: {
    params: {
      movie: MovieItem;
    }
  }
};
