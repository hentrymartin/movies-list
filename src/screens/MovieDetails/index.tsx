import React from 'react';
import { observer } from 'mobx-react-lite';

import {
  MovieDetailsWrapper,
  MovieDetailsTitle,
  MovieDescription,
  MenuWrapper,
} from './MovieDetails.styles';
import { MovieDetailsProps } from './MovieDetails.types';
import Image from '../../components/MovieImage';
import PopularSection from '../../components/PopularSection';

/**
 * This shows the movie details form route params
 */
const MovieDetails = observer(({ route }: MovieDetailsProps) => {
  const { params:  { movie }} = route;

  return (
    <MovieDetailsWrapper>
      <MovieDetailsTitle testID="movie-title">{movie.title}</MovieDetailsTitle>

      <Image uri={`http://image.tmdb.org/t/p/original/${movie.backdrop_path}`}/>

      <MenuWrapper>
        <PopularSection movie={movie} />
      </MenuWrapper>
      
      <MovieDescription>
        {movie.overview}
      </MovieDescription>
    </MovieDetailsWrapper>
  )
});

export default MovieDetails;
