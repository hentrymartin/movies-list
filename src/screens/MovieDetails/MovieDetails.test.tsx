import React from 'react';
import { render } from '@testing-library/react-native';
import MockedNavigator from '../../utils/testing/mockedNavigator';
import MovieDetails from '.';
import { Movie } from '../../mocks/movies';

describe('<MovieDetails />', () => {
  it('check if it is rendered properly', () => {
    const route = {
      params: {
        movie: Movie,
      },
    };
    const { toJSON } = render(
      <MockedNavigator component={() => <MovieDetails route={route}/>} />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
