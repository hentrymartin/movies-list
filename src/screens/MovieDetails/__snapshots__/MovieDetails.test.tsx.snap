// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieDetails /> check if it is rendered properly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          false,
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                MockedScreen
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "paddingBottom": 10,
                            "paddingLeft": 0,
                            "paddingRight": 0,
                            "paddingTop": 10,
                          },
                        ]
                      }
                    >
                      <Text
                        style={
                          Array [
                            Object {
                              "fontSize": 20,
                              "fontWeight": "500",
                              "marginBottom": 10,
                              "marginLeft": 0,
                              "marginRight": 0,
                              "marginTop": 10,
                              "textAlign": "center",
                            },
                          ]
                        }
                        testID="movie-title"
                      >
                        Darlin'
                      </Text>
                      <Image
                        resizeMode="contain"
                        source={
                          Object {
                            "uri": "http://image.tmdb.org/t/p/original//k60vmF4t3X9Yw4oAiQ2s0KtN7RU.jpg",
                          }
                        }
                        style={
                          Array [
                            Object {
                              "height": 250,
                            },
                          ]
                        }
                      />
                      <View
                        style={
                          Array [
                            Object {
                              "alignItems": "flex-end",
                              "display": "flex",
                              "justifyContent": "center",
                              "paddingBottom": 0,
                              "paddingLeft": 20,
                              "paddingRight": 20,
                              "paddingTop": 0,
                            },
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "flexDirection": "row",
                                "marginTop": 10,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "flexBasis": 0,
                                  "flexGrow": 1,
                                  "flexShrink": 1,
                                },
                              ]
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "fontSize": 16,
                                  },
                                ]
                              }
                            >
                              Popolarity
                            </Text>
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "black",
                                    "fontSize": 20,
                                    "fontWeight": "500",
                                  },
                                ]
                              }
                            >
                              136.328
                            </Text>
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "flexBasis": 0,
                                  "flexGrow": 1,
                                  "flexShrink": 1,
                                },
                              ]
                            }
                          >
                            <Text
                              style={
                                Array [
                                  Object {
                                    "fontSize": 16,
                                  },
                                ]
                              }
                            >
                              Vote
                            </Text>
                            <Text
                              ratingLevel="bad"
                              style={
                                Array [
                                  Object {
                                    "color": "red",
                                    "fontSize": 20,
                                    "fontWeight": "500",
                                  },
                                ]
                              }
                            >
                              4
                            </Text>
                          </View>
                          <View
                            accessible={true}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                            testID="favourite-icon"
                          >
                            <Image
                              source={
                                Object {
                                  "process": [Function],
                                }
                              }
                              style={
                                Array [
                                  Object {
                                    "height": 40,
                                    "width": 40,
                                  },
                                ]
                              }
                              testID="star-icon"
                            />
                          </View>
                        </View>
                      </View>
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "grey",
                              "lineHeight": 22,
                              "paddingBottom": 10,
                              "paddingLeft": 20,
                              "paddingRight": 20,
                              "paddingTop": 10,
                            },
                          ]
                        }
                      >
                        Found at a Catholic hospital filthy and ferocious, feral teenager Darlin’ is whisked off to a care home run by The Bishop and his obedient nuns, where she’s to be rehabilitated into a "good girl" as an example of the miraculous work of the church. But Darlin’ holds a secret darker than the "sins" she is threatened with, and she is not traveling alone. The Woman who raised her, equally fierce and feral, is ever present in the shadows of Darlin’s psyche and is determined to come for her no matter who tries to get in her way.
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
